File: java/leetcode/CardFlipping.java
//Leetcode problem  822 Card Flipping Game
//Solution written by Xuqiang Fang on 31 May, 2018
import java.util.ArrayList;
import java.util.List;
import java.util.HashMap;
import java.util.Map;
import java.util.HashSet;
import java.util.Set;
import java.util.PriorityQueue;
import java.util.Stack;
import java.util.Queue;
import java.util.LinkedList;
import java.util.Arrays;
class Solution{
    public int flipgame(int[] f, int[] b){
        int[] a = new int[2001];
        int n = f.length;
        for(int i=0; i<n; ++i){
            if(f[i] == b[i]){
                a[f[i]] = Integer.MIN_VALUE;
            }
            else{
                a[f[i]]++;
                a[b[i]]++;
            }
        }
        for(int i=1; i<=2000; ++i){
            if(a[i] >= 1){
                return i;
            }
        }
        return 0;
    }
}

public class CardFlipping{
	public static void main(String[] args){
		Solution s = new Solution();
        int[] f = {1,2,4,4,7};
        int[] b = {1,3,4,1,3};
        System.out.println(s.flipgame(f,b));
	}
}

File: cpp/leetcode/CardFlipping.cpp
//Leetcode Problem No  822 Card Flipping Game
//Solution written by Xuqiang Fang on 31 May, 2018

#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <unordered_map>
#include <unordered_set>
#include <stack>
#include <queue>

using namespace std;
class Solution{
public:
    int flipgame(vector<int>& fronts, vector<int>& backs) {
        vector<int> a(2001,0);    
        int n = fronts.size();
        for(int i=0; i<n; ++i){
            if(fronts[i] != backs[i]){
                a[fronts[i]]++;
                a[backs[i]]++;
            }
            else{
                a[fronts[i]] = INT_MIN;
            }
        }
        for(int i=1; i<=2000; ++i){
            if(a[i] >= 1 ){
                return i;
            }
        }
        return 0;
    }
};

int main(){
    Solution s;
    vector<int> f{1,2,4,4,7};
    vector<int> b{1,3,4,1,3};
    cout << s.flipgame(f, b) << endl;
    vector<int> fr{1,1};
    vector<int> ba{2,2};
    cout << s.flipgame(fr, ba) << endl;
    return 0;
}
